contracts/gas-service/AxelarGasService.sol

```diff
@@ -38,6 +50,58 @@ contract AxelarGasService is Upgradable, IAxelarGasService {
         _;
     }
 
+    /**
+     * @notice Pay for gas for any type of contract execution on a destination chain.
+     * @dev This function is called on the source chain before calling the gateway to execute a remote contract.
+     * @dev If estimateOnChain is true, the function will estimate the gas cost and revert if the payment is insufficient.
+     * @param sender The address making the payment
+     * @param destinationChain The target chain where the contract call will be made
+     * @param destinationAddress The target address on the destination chain
+     * @param payload Data payload for the contract call
+     * @param executionGasLimit The gas limit for the contract call
+     * @param estimateOnChain Flag to enable on-chain gas estimation
+     * @param refundAddress The address where refunds, if any, should be sent
+     * @param params Additional parameters for gas payment
+     */
+    function payGas(
+        address sender,
+        string calldata destinationChain,
+        string calldata destinationAddress,
+        bytes calldata payload,
+        uint256 executionGasLimit,
+        bool estimateOnChain,
+        address refundAddress,
+        bytes calldata params
+    ) external payable override {
+        GasPaymentType gasPaymentType = GasPaymentType.NativeForContractCall;
+
+        if (params.length >= 32) {
+            (gasPaymentType) = abi.decode(params, (GasPaymentType));
+        }
+
+        if (estimateOnChain) {
+            uint256 gasEstimate = estimateGasFee(
+                destinationChain,
+                destinationAddress,
+                payload,
+                executionGasLimit,
+                gasPaymentType >= GasPaymentType.NativeForExpressCall && gasPaymentType <= GasPaymentType.TokenForExpressCallWithToken
```

Comment:
add `params` field to `estimateGasFee` for future proofing. Should ignore it for now